### PR TITLE
Change Homepage Sorting to Nearest Eatery

### PR DESCRIPTION
**Pull request by rs929**
## Overview

Implemented Homepage sorting to eatery

### Change 1

- Removed nearest eateries carousel 
- Implemented homepage sorting to nearest eateries

## Test Coverage

- Tested accurately with current location in the simulator

## Screenshots (optional)

<details>
  <summary>TODO</summary>
  <table>
    <tr>
      <td>Before</td>
      <td>After</td>
    </tr>
  <tr>
    <td><video src="enterurlhere" type="video/mp4"></td>
    <td><video src="enterurlhere" type="video/mp4"></td>
  </tr>
 </table>
</details>

<details>
  <summary>TODO</summary>
  <table>
    <tr>
      <td>Before</td>
      <td>After</td>
    </tr>
  <tr>
    <td></td>
    <td></td>
  </tr>
 </table>
</details>

<details>
  <summary>TODO</summary>
  
  <video src="enterurlhere" type="video/mp4">
</details>

### CODE REVIEW

File: Eatery Blue/UI/EateryCard/EateryLargeCardContentView.swift [hunk 1]
```diff
@@ -15,12 +17,16 @@ class EateryLargeCardContentView: UIView {
     private let imageView = UIImageView()
     private let imageTintView = UIView()
     private let alertsStackView = UIStackView()
+
```
Comment (MrPeterss) [inline Eatery Blue/UI/EateryCard/EateryLargeCardContentView.swift:20]: extra new line?

File: Eatery Blue/UI/EateryCard/EateryLargeCardContentView.swift [hunk 2]
```diff
@@ -7,6 +7,8 @@
 
 import EateryModel
 import UIKit
+import Combine
+
```
Comment (MrPeterss) [inline Eatery Blue/UI/EateryCard/EateryLargeCardContentView.swift:11]: extra newline?

File: Eatery Blue/UI/EateryCard/EateryLargeCardContentView.swift [hunk 3]
```diff
@@ -15,12 +17,16 @@ class EateryLargeCardContentView: UIView {
...
+    
 
     private let labelStackView = UIStackView()
     private let titleLabel = UILabel()
     private let subtitleLabels = [UILabel(), UILabel()]
     private let favoriteButton = ButtonView(content: UIView())
     private let favoriteButtonImage = UIImageView()
+    
+    private var cancellables = Set<AnyCancellable>()
+
```
Comment (MrPeterss) [inline Eatery Blue/UI/EateryCard/EateryLargeCardContentView.swift:29]: one of these has to be extra

File: Eatery Blue/UI/EateryCard/EateryLargeCardContentView.swift [hunk 4]
```diff
@@ -147,7 +153,19 @@ class EateryLargeCardContentView: UIView {
...
+                    eatery,
+                    style: .medium,
+                    font: .preferredFont(for: .footnote, weight: .medium),
+                    userLocation: userLocation,
+                    date: Date()
+                ).first
+            }
+            .store(in: &cancellables)
+        
+//        subtitleLabels[1].attributedText = EateryFormatter.default.eateryCardFormatter(eatery, date: Date())
```
Comment (MrPeterss) [inline Eatery Blue/UI/EateryCard/EateryLargeCardContentView.swift:168]: lets just remove this

File: Eatery Blue/UI/HomeScreen/HomeModelController.swift [hunk 5]
```diff
@@ -169,7 +169,7 @@ class HomeModelController: HomeViewController {
         carouselView.layoutMargins = UIEdgeInsets(top: 0, left: 16, bottom: 0, right: 16)
         return carouselView
     }
-
+
```
Comment (MrPeterss) [inline Eatery Blue/UI/HomeScreen/HomeModelController.swift:172]: extra tab

File: Eatery Blue/UI/HomeScreen/HomeModelController.swift [hunk 6]
```diff
@@ -206,7 +206,21 @@ class HomeModelController: HomeViewController {
...
+                let dist1 = eatery1.walkTime(userLocation: userLocation)
+                let dist2 = eatery2.walkTime(userLocation: userLocation)
+                guard let dist1 else { return true }
+                guard let dist2 else { return true }
+                return dist1 < dist2
+            })
+        }
+        .store(in: &cancellables)
+
+
```
Comment (MrPeterss) [inline Eatery Blue/UI/HomeScreen/HomeModelController.swift:223]: extra newline

File: Eatery Blue/UI/HomeScreen/HomeModelController.swift [hunk 7]
```diff
@@ -289,7 +304,6 @@ class HomeModelController: HomeViewController {
     @objc func refreshFavorites(_ notification: Notification) {
         updateCellsFromState()
     }
-
```
Comment (MrPeterss) [inline Eatery Blue/UI/HomeScreen/HomeModelController.swift:292]: this line should be here

File: Eatery Blue/main.swift [hunk 8]
```diff
@@ -67,7 +67,6 @@ struct EateryBlue: ParsableCommand {
 
         AppDelegate.main()
     }
-
```
Comment (MrPeterss) [inline Eatery Blue/main.swift:70]: put this line back too

File: Eatery Blue/UI/Carousel/CarouselMoreEateriesCollectionViewCell.swift [hunk 9]
```diff
@@ -69,5 +69,4 @@ class CarouselMoreEateriesCollectionViewCell: UICollectionViewCell {
             make.top.greaterThanOrEqualToSuperview()
         }
     }
-
```
Comment (MrPeterss) [inline Eatery Blue/UI/Carousel/CarouselMoreEateriesCollectionViewCell.swift:72]: put this line back

File: Eatery Blue/UI/HomeScreen/HomeModelController.swift [hunk 10]
```diff
@@ -206,7 +206,21 @@ class HomeModelController: HomeViewController {
...
-        let openEateries = currentEateries.filter(\.isOpen)
+        var openEateries = currentEateries.filter(\.isOpen)
+        
+        LocationManager.shared.$userLocation
+        .sink { userLocation in
+            openEateries = openEateries.sorted(by: { eatery1, eatery2 in
+                let dist1 = eatery1.walkTime(userLocation: userLocation)
+                let dist2 = eatery2.walkTime(userLocation: userLocation)
+                guard let dist1 else { return true }
+                guard let dist2 else { return true }
```
Comment (MrPeterss) [inline Eatery Blue/UI/HomeScreen/HomeModelController.swift:217]: New line should go after the guard. Also, I believe lines 216 and 217 can be condensed to:
`guard let dist1, let dist2 else { return true }`

**Review by MrPeterss — CHANGES_REQUESTED** [review]
@Xhether nice work, Charles! Overall really clean pr. Just make sure to fix some of the styling changes that I mentioned. In general, we want to put new lines at the beginning and end of classes (after the first curly bracket, and before the last curly bracket). Also, whenever you are using brackets in your logic (for if statements, completion handlers, etc), we want to put new lines after that to break it up (this includes after guard statements).

**Review by MrPeterss — APPROVED** [review]
Nice work!

**Review by Xhether — APPROVED** [review]
yippee

**Pull request merged**